deepchem/feat/base_classes.py

```diff
@@ -289,12 +291,37 @@ def featurize(self, structures: Iterable[JSON],
     return features
 
 
-class CompositionFeaturizer(Featurizer):
+  def _featurize(self, structure):
+    """Calculate features for a single crystal structure.
+
+    Parameters
+    ----------
+    structure: pymatgen.Structure object
+      Structure object with 3D coordinates and periodic lattice.
+
+    """
+
+    raise NotImplementedError('Featurizer is not defined.')
+
+  def __call__(self, structures: Iterable[Dict[str, Any]]):
+    """Calculate features for crystal structures.
+
+    Parameters
+    ----------
+    structures: Iterable[Dict[str, Any]]
+      An iterable of pymatgen.Structure dictionaries.
+
+    """
+
+    return self.featurize(structures)
```

Comment:
This override is not necessary. I think the classes in base_classes.py should just override `featurize` basically.

If you want to add these lines and specify arguments  of `__call__` or `_featurize`, this featurizer no longer needs to inherit `Featurizer`.